src/solvers/refinement/string_refinement_util.cpp

```diff
@@ -176,6 +176,18 @@ interval_sparse_arrayt interval_sparse_arrayt::of_array_list(
   return sparse_array;
 }
 
+optionalt<interval_sparse_arrayt>
+interval_sparse_arrayt::of_expr(const exprt &expr, const exprt &extra_value)
+{
+  if(const auto &array_expr = expr_try_dynamic_cast<array_exprt>(expr))
+    return interval_sparse_arrayt(*array_expr, extra_value);
+  if(const auto &with_expr = expr_try_dynamic_cast<with_exprt>(expr))
+    return interval_sparse_arrayt(*with_expr);
+  if(expr.id() == "array-list")
```

Comment:
A proper irep_id should be added for this

Reply:
If that's OK I'd prefer to delay that to anoter PR (prepared here https://github.com/romainbrenguier/cbmc/pull/7) because it will require other reviewers as it touch other parts of CBMC as well.